src/Aspire.Dashboard/Components/Pages/StructuredLogs.razor.cs

```diff
@@ -221,7 +221,16 @@ private void UpdateApplications()
     {
         _applications = TelemetryRepository.GetApplications();
         _applicationViewModels = ApplicationsSelectHelpers.CreateApplications(_applications);
-        _applicationViewModels.Insert(0, _allApplication);
+
+        if (_applicationViewModels.Count != 1)
+        {
+            _applicationViewModels.Insert(0, _allApplication);
+        }
+        else
+        {
+            PageViewModel.SelectedApplication = _applicationViewModels.Single();
+            NavigationManager.NavigateTo(GetUrlFromSerializableViewModel(ConvertViewModelToSerializable()));
+        }
```

Comment:
No, this is unnecessary. The resource selector on this page is a filter. A filter of `All` and selecting the only option are the same.

This change is only for console logs and metrics where no data is displaying if `None` is selected.